src/wallet/wallet.cpp

```diff
@@ -1586,7 +1586,9 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
     }
     double progress_current = progress_begin;
     while (block_height && !fAbortRescan && !chain().shutdownRequested()) {
-        m_scanning_progress = (progress_current - progress_begin) / (progress_end - progress_begin);
+        if (progress_end - progress_begin > 0.0) {
+            m_scanning_progress = (progress_current - progress_begin) / (progress_end - progress_begin);
+        }
         if (*block_height % 100 == 0 && progress_end - progress_begin > 0.0) {
```

Comment:
It seems to me there is no need to check twice that progress_end - progress_begin is greater than zero, even if the compiler is clever enough to optimise it. Perhaps:
```diff
-        m_scanning_progress = (progress_current - progress_begin) / (progress_end - progress_begin);
-        if (*block_height % 100 == 0 && progress_end - progress_begin > 0.0) {
-            ShowProgress(strprintf("%s " + _("Rescanning...").translated, GetDisplayName()), std::max(1, std::min(99, (int)(m_scanning_progress * 100))));
+        if (progress_end > progress_begin) {
+            m_scanning_progress = (progress_current - progress_begin) / (progress_end - progress_begin);
+            if (*block_height % 100 == 0) {
+                ShowProgress(strprintf("%s " + _("Rescanning...").translated, GetDisplayName()), std::max(1, std::min(99, (int)(m_scanning_progress * 100))));
+            }
```
or
```diff
+        double blocks_to_scan = (progress_end - progress_begin);
+        if (blocks_to_scan > 0.0) {
+            m_scanning_progress = (progress_current - progress_begin) / blocks_to_scan;
+            if (*block_height % 100 == 0) {
+                ShowProgress(strprintf("%s " + _("Rescanning...").translated, GetDisplayName()), std::max(1, std::min(99, (int)(m_scanning_progress * 100))));
+            }
```

Reply:
I wanted to keep the diff small by touch as few lines as possible to make review trivial :)